zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -481,6 +508,10 @@ where
         sapling_shielded_data: &Option<sapling::ShieldedData<AnchorV>>,
     ) -> Result<(), ValidateContextError> {
         if let Some(sapling_shielded_data) = sapling_shielded_data {
+            for cm_u in sapling_shielded_data.note_commitments() {
```

Comment:
Missing docs: let's explain why we can't remove the sapling notes in `revert_chain_state_with`

Reply:
Added in dce91134